coderd/provisionerdaemons.go

```diff
@@ -18,31 +19,50 @@ import (
 // @Produce json
 // @Tags Provisioning
 // @Param organization path string true "Organization ID" format(uuid)
+// @Param limit query int false "Page limit"
+// @Param ids query []string false "Filter results by job IDs" format(uuid)
+// @Param status query codersdk.ProvisionerJobStatus false "Filter results by status" enums(pending,running,succeeded,canceling,canceled,failed)
 // @Param tags query object false "Provisioner tags to filter by (JSON of the form {'tag1':'value1','tag2':'value2'})"
 // @Success 200 {array} codersdk.ProvisionerDaemon
 // @Router /organizations/{organization}/provisionerdaemons [get]
 func (api *API) provisionerDaemons(rw http.ResponseWriter, r *http.Request) {
 	var (
-		ctx      = r.Context()
-		org      = httpmw.OrganizationParam(r)
-		tagParam = r.URL.Query().Get("tags")
-		tags     = database.StringMap{}
-		err      = tags.Scan([]byte(tagParam))
+		ctx = r.Context()
+		org = httpmw.OrganizationParam(r)
 	)
 
-	if tagParam != "" && err != nil {
+	qp := r.URL.Query()
+	p := httpapi.NewQueryParamParser()
+	limit := p.PositiveInt32(qp, 50, "limit")
+	ids := p.UUIDs(qp, nil, "ids")
+	tagsRaw := p.String(qp, "", "tags")
+	p.ErrorExcessParams(qp)
+	if len(p.Errors) > 0 {
 		httpapi.Write(ctx, rw, http.StatusBadRequest, codersdk.Response{
-			Message: "Invalid tags query parameter",
-			Detail:  err.Error(),
+			Message:     "Invalid query parameters.",
+			Validations: p.Errors,
 		})
 		return
 	}
 
+	tags := database.StringMap{}
+	if tagsRaw != "" {
+		if err := tags.Scan([]byte(tagsRaw)); err != nil {
+			httpapi.Write(ctx, rw, http.StatusBadRequest, codersdk.Response{
+				Message: "Invalid tags query parameter",
+				Detail:  err.Error(),
+			})
+			return
+		}
+	}
+
```

Comment:
non-blocking, suggestion: `qp.StringMap` could be a nice QoL feature to have here!

Reply:
Ended up implementing that here: https://github.com/coder/coder/pull/16578